internal/install/recipe_installer.go

```diff
@@ -165,45 +183,43 @@ func (i *RecipeInstaller) Install() error {
 	// Report discovered recipes as available
 	i.status.ReportRecipesAvailable(recipesForReport)
 
-	log.Debugf("InstallerContext: %+v", i.InstallerContext)
-	log.WithFields(log.Fields{
-		"ShouldPrompt":    i.ShouldPrompt(),
-		"RecipesProvided": i.RecipesProvided(),
-	}).Debug("flag summary")
-
 	var entityGUID string
 	if !i.RecipesProvided() {
 
 		if i.SkipInfraInstall {
 			i.status.ReportRecipeSkipped(execution.RecipeStatusEvent{Recipe: *infraAgentRecipe})
 		} else {
-			log.Debugf("Installing infrastructure agent")
-			entityGUID, err = i.executeAndValidateWithProgress(m, infraAgentRecipe)
-			if err != nil {
-				log.Error(i.failMessage(infraAgentRecipeName))
-				return i.fail(err)
+			if i.ShouldInstallInfraAgent() {
```

Comment:
Is this necessary within the current block?  `SkipInfraInstall` has already evaluated to `false` here.  It's a bit confusing that this needs to be checked again.  Maybe a rename of the method would help with clarity?

Reply:
Yeah, I see how that's a little confusing, though I think the logic is accurate in the case where a recipe is installed.  The `Should` method checks if there were recipes provided too, as well as the skip flag.  So in the case of the skip flag, we want to report skipped, but in the case a recipe (name or path) was provided, I don't think we do.  I suppose if it didn't matter, the `if` at the top of the block could just check `ShouldInstall...()` and then the `else` would report the skip.  What do you think?

Reply:
LGTM.